contrib/k8s/README.md

```diff
@@ -1,41 +1,123 @@
-# dex
+# Deploying dex on Kubernetes
 
-## Getting Started
+This document will allow you to set up dex in your Kubernetes; the configuration files used are generally useful, but will need to be modified to meet the needs of your deployment. The places that are likely to need modification will be called out as often as possible in this document.
 
-**Warning**: Hacks Ahead.
+## Prerequisites and Assumptions
 
-You must be running cluster wide DNS for this to work.
+The document assumes that you already have a cluster with at least one worker up and running. The easiest way to bring up a small cluster for experimentation is the [coreos-kubernetes single node](coreos-kubernetes-single-node) Vagrant installer.
 
-Install your dockercfg. There is no nice way to do this:
+The other assumption is that your Kubernetes cluster will be routable on `172.17.4.99`  (which is what it will be if you use [coreos-kubernetes single node][coreos-kubernetes-single-node], and the issuer URL for your dex installation is `http://172.17.4.99:30556`; in production installations you will need to make sure that you are serving on https and you will likely want a host name on the normal TLS port instead of an IP address.
```

Comment:
Nit: "you will likely want a host name on the normal TLS port instead of an IP address" -- this is a little unclear. Maybe something along the lines of "you will likely want to use a hostname rather than an IP address"